### PR TITLE
Fail early if the volume expansion request came for static PVs.

### PR DESCRIPTION
**Pull request by humblec**
if the request comes in for statically provisioned PV, the
driver does not know the credentials to fetch for reaching
out to the expansion server. In this cases, we should fail
early instead of trying it.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -1187,6 +1187,14 @@ func (plugin *glusterfsPlugin) ExpandVolumeDevice(spec *volume.Spec, newSize res
 	if err != nil {
 		return oldSize, fmt.Errorf("failed to get volumeID for volume %s: %v", volumeName, err)
 	}
+
+	if volumeID == volumeName {
+		// If the volume was provisioned/created statically
```
Comment (jsafrane) [inline pkg/volume/glusterfs/glusterfs.go]: The comment and the error message is misleading. Even a "static PV", i.e. a PV created by cluster admin, can have valid volume ID and be resizeable.

IMO you should check if the volume has enough info to resize it and report what field / annotation is missing.

**Comment by dims** [comment]
@humblec @jsafrane do we need this for 1.17?

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (humblec) [reply pkg/volume/glusterfs/glusterfs.go]: @jsafrane If the volume is statically provisioned, we miss the annotation in pv spec and its hard to judge how admin provisioned the volume and from which server. Thats why we are skipping the resize on it.

Comment (jsafrane) [reply pkg/volume/glusterfs/glusterfs.go]: Again, even a "static PV" can have valid volume ID annotation and be resizeable. Admin can create such PV easily.

You should report that "PV misses annotation XYZ and cannot be resized". And the comment should reflect it.

**Comment by humblec** [comment]
@dims I would have been happy to get it in, however have to think more as @jsafrane mentioned in last comment.

**Pull request closed without merging**